lua/nvim-tree/explorer/filters.lua

```diff
@@ -108,29 +136,29 @@ end
 
 ---@param path string
 ---@return boolean
-local function custom(path)
-  if not M.config.filter_custom then
+local function custom(self, path)
+  if not self.config.filter_custom then
     return false
   end
 
   local basename = utils.path_basename(path)
 
   -- filter user's custom function
-  if M.custom_function and M.custom_function(path) then
+  if self.custom_function and self:custom_function(path) then
```

Comment:
I don't mind. You can rewrite it completely if you like ;) 
I'm not particularly attached to my code. 
I'm more interested in the final result and the functionality we can deliver.
That was a bug, and you fixed it, so that's perfect. Feel free to commit your changes directly to the branch if you like.